crates/biome_service/src/workspace.rs

```diff
@@ -403,9 +419,16 @@ pub enum FeatureKind {
     OrganizeImports,
     Search,
     Assists,
+    SyntaxTree,
```

Comment:
Syntax and formatter IR aren't needed. If a file can be handled, it can be parsed, hence it can emit a CST. If a file can be formatted, it can emit a IR.

Control flow is a black sheep. I suppose we can add it

Reply:
The debug methods can technically be unimplemented, and that leads to throwing errors in the playground. 

For example, the html file handler is currently in this state:

https://github.com/biomejs/biome/blob/1792846b4d694d06f6d3c0e399b730e59490ded5/crates/biome_service/src/file_handlers/html.rs#L59-L83

Reply:
> The debug methods can technically be unimplemented

That is true for all methods, and it should be the **the client** to deal with that. That's a design decision because various clients, even servers, can consume the Workspace. When used by servers, all methods can technically fail, that's why even methods that don't need to return `Result`, they do.

Unfortunately, JS doesn't provide any signature for when a method can throw an error at runtime, so we are stuck with what we have.

Regarding the `FeatureKind`. I think we should reduce the proposed changes to a single `FeatureKind::Debug`, because that's inline with what we have today. We have `::Format`, not `::Formating`/`::FormatOnType`/`::FormatRange`. It will be up to us to implement all the debugging features at once.